crates/webidl/src/util.rs

```diff
@@ -693,6 +638,149 @@ impl<'src> FirstPassRecord<'src> {
             }
         }
     }
+
+    pub fn create_imports(
+        &self,
+        kind: backend::ast::ImportFunctionKind,
+        id: &OperationId<'src>,
+        data: &OperationData2<'src>,
+    )
+        -> Vec<backend::ast::ImportFunction>
+    {
+        // First up expand all the signatures in `data` into all signatures that
+        // we're going to generate. These signatures will be used to determine
+        // the names for all the various functions.
+        #[derive(Clone)]
+        struct ExpandedSig<'a> {
+            orig: &'a Signature<'a>,
+            args: Vec<IdlType<'a>>,
+        }
+
+        let mut actual_signatures = Vec::new();
+        'outer:
+        for signature in data.signatures.iter() {
+            let start = actual_signatures.len();
+
+            // Start off with an empty signature, this'll handle zero-argument
+            // cases and otherwise the loop below will continue to add on to this.
+            actual_signatures.push(ExpandedSig {
+                orig: signature,
+                args: Vec::with_capacity(signature.args.len()),
+            });
+
+            for (i, arg) in signature.args.iter().enumerate() {
+                let cur = actual_signatures.len();
+                // If any argument in this signature can't be converted we have
+                // to throw out the entire signature, so revert back to the
+                // beginning and then keep going.
+                let idl_type = match arg.ty.to_idl_type(self) {
+                    Some(t) => t,
+                    None => {
+                        actual_signatures.truncate(start);
+                        continue 'outer
+                    }
+                };
+
+                // The first arugment gets pushed directly in-place, but all
+                // future expanded arguments will cause new signatures to be
+                // created. If we have an optional argument then we consider the
+                // already existing signature as the "none" case and the flatten
+                // below will produce the "some" case, so we've already
+                // processed the first argument effectively.
+                let mut first = !arg.optional;
```

Comment:
the `first` variable = whether or not this is in the set of arguments that all signatures for this method share, right?

The comment doesn't quite match that, and says "the first argument" blah blah, when it is really "the set of common arguments" blah blah, right? We should definitely clarify this for future readers!

Reply:
Readign some more, I think I am misunderstanding what this loop does, because what I thoguth it does is handled below. Now I am not really sure what is up with this.

Reply:
Nah your first comment makes sense! I think I've clarified it now though